### PR TITLE
Expose ComputePlatformResolvedLocale

### PR DESCRIPTION
**Pull request by GaryQian**
Exposes `ComputePlatformResolvedLocale` for https://github.com/flutter/flutter/issues/47087

Relevant code added in flutter/engine#18645 and flutter/engine#19266

### CODE REVIEW

**Comment by GaryQian** [comment]
Will hold off on this until the TODOs introduced in https://github.com/flutter/engine/pull/18645 are resolved to ensure the feature produces the correct values.

**Comment by GaryQian** [comment]
This method is a direct call to dart:ui's Window.computePlatformResolvedLocale and is tested at the engine level of each platform.

File: packages/flutter/lib/src/widgets/binding.dart [hunk 1]
```diff
@@ -944,6 +944,46 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
...
+  /// is returned. See:
+  /// https://developer.apple.com/documentation/foundation/nsbundle/1417249-preferredlocalizationsfromarray?language=objc
+  /// for details on the used method.
+  ///
+  /// iOS treats script code as necessary for a match, so a user preferred locale of
+  /// `zh_Hans_CN` will not resolve to a supported locale of `zh_CN`.
+  ///
+  /// Since implemntation may vary by platform and has potential to be heavy,
+  /// it is recommended to cache the results of this method if the value if it
+  /// is used multiple times.
```
Comment (justinmc) [inline packages/flutter/lib/src/widgets/binding.dart:979]:
```suggestion
  /// it is recommended to cache the results of this method if the value is
  /// used multiple times.
```

**Review by justinmc — APPROVED** [review]
LGTM

CC @shihaohong in case this is relevant.

**Pull request merged**